api/rest/restcore/issues_rest.php

```diff
@@ -370,3 +374,59 @@ function rest_issue_monitor_add( \Slim\Http\Request $p_request, \Slim\Http\Respo
 	return $p_response->withStatus( HTTP_STATUS_CREATED, "Users are now monitoring issue $t_issue_id" )->
 		withJson( array( 'issues' => array( $t_issue ) ) );
 }
+
+/**
+ * Get files associated with an issue.
+ *
+ * @param \Slim\Http\Request $p_request   The request.
+ * @param \Slim\Http\Response $p_response The response.
+ * @param array $p_args Arguments
+ * @return \Slim\Http\Response The augmented response.
+ */
+function rest_issue_files_get( \Slim\Http\Request $p_request, \Slim\Http\Response $p_response, array $p_args ) {
+	$t_issue_id = $p_args['id'];
+	$t_file_id = isset( $p_args['file_id'] ) ? $p_args['file_id'] : null;
+
+	$t_data = array(
+		'query' => array( 'issue_id' => $t_issue_id, 'file_id' => $t_file_id )
+	);
+
+	$t_command = new IssueFileGetCommand( $t_data );
+	$t_internal_files = $t_command->execute();
+
+	$t_files = array();
+	foreach( $t_internal_files as $t_internal_file ) {
+		$t_file = array(
+			'id' => (int)$t_internal_file['id'],
+			'filename' => $t_internal_file['display_name'],
+			'reporter' => mci_account_get_array_by_id( $t_internal_file['user_id'] ),
+			'size' => (int)$t_internal_file['size'],
+			'created_at' => ApiObjectFactory::datetimeString( $t_internal_file['date_added'] ),
+			'icon' => $t_internal_file['icon']['url'],
+			'alt' => $t_internal_file['alt'],
+			'access' => array(
+				'download' => $t_internal_file['can_download'],
+				'delete' => $t_internal_file['can_delete']
+			)
+		);
+
+		if( $t_internal_file['can_download'] ) {
+			$t_file['download_url'] = $t_internal_file['download_url'];
+		}
+
+		if( $t_internal_file['can_delete'] ) {
+			$t_file['delete_url'] = $t_internal_file['delete_url'];
+		}
```

Comment:
I did not expect to see this. I thought deleting a file is another REST api, no?

Reply:
> I did not expect to see this. I thought deleting a file is another REST api, no?

Yes, there is another REST API for deleting a file.  The URL here is for the web URL for the delete page for use in the web UI.  I can see how this can be confusing.  Will look again and re-consider.

Reply:
@syncguru Removed the `download_url` and `delete-url`.